### PR TITLE
Clarify why posts deleted by Data Retention job are visible in the webapp

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Posts that got deleted by the Data Retention job are still shown in the UI, until the user refreshes the webapp. The user might also need to clear their browser cache in order to see the posts.

#### Ticket Link
https://mattermost.zendesk.com/agent/tickets/42430

### CODE REVIEW

**Comment by hanzei** [comment]
Unrelated to my PR: The docs on this page talk about a "deletion job". I suppose they mean "data retention job", right?

File: source/comply/data-retention-policy.rst [hunk 1]
```diff
@@ -63,6 +63,11 @@ Replies that did not exceed the message duration are still displayed in the user
 
 If there was a file attached to the message, it will be removed from the user interface only.  
 
+Why can I still see messages that were supposedly deleted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (cwarnermm) [inline source/comply/data-retention-policy.rst:67]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by hanzei** [comment]
@cwarnermm what do you think about my question above? https://github.com/mattermost/docs/pull/6823#issuecomment-1847031857

**Review by BenCookie95 — APPROVED** [review]

File: source/comply/data-retention-policy.rst [hunk 2]
```diff
@@ -63,6 +63,11 @@ Replies that did not exceed the message duration are still displayed in the user
 
 If there was a file attached to the message, it will be removed from the user interface only.  
 
+Why can I still see messages that were supposedly deleted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The web and desktop app cache posts are cached. Posted deleted by a data retention job will be visible to end users until they clear their cache and refresh.
```
Comment (hanzei) [inline source/comply/data-retention-policy.rst:69]:
```suggestion
The web and desktop app cache posts are cached. Posts deleted by a data retention job will be visible to end users until they clear their cache and refresh.
```

File: source/comply/data-retention-policy.rst [hunk 3]
```diff
@@ -63,6 +61,11 @@ Replies that did not exceed the message duration are still displayed in the user
 
 If there was a file attached to the message, it will be removed from the user interface only.  
 
+Why can I still see messages that were supposedly deleted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The web and desktop app cache posts are cached. Posts deleted by a data retention job will be visible to end users until they clear their cache and refresh.
```
Comment (hanzei) [inline source/comply/data-retention-policy.rst:67]:
```suggestion
The web and desktop app cache posts. Posts deleted by a data retention job will be visible to end users until they clear their cache and refresh.
```

**Review by cwarnermm — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]

**Pull request merged**